lib/pure/os.nim

```diff
@@ -3217,6 +3184,43 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo {.noWeirdTarget.
         raiseOSError(osLastError(), path)
     rawToFormalFileInfo(rawInfo, path, result)
 
+proc sameFileContent*(path1, path2: string; checkSize = false; bufferSize = 8192.Positive): bool {.
+    rtl, extern: "nos$1", tags: [ReadIOEffect], noWeirdTarget.} =
+  ## Returns true if both pathname arguments refer to files with identical
+  ## binary content.
+  ##
+  ## If `checkSize` is `true` then checks the file sizes *before reading the files*,
+  ## if the files have different file sizes they can not have the same contents,
+  ## `checkSize = true` may be faster, specially for very big files.
+  ##
+  ## See also:
+  ## * `sameFile proc <#sameFile,string,string>`_
+  var
+    a, b: File
+  if not open(a, path1): return false
+  if not open(b, path2):
+    close(a)
+    return false
+  if checkSize and getFileInfo(a).size != getFileInfo(b).size: return false
```

Comment:
leaks a file descriptor here! This is such a frequent error, I see it all the time.
this is why `defer` should be used more widely IMO pending benchmarks to do refs https://forum.nim-lang.org/t/4022#44758

Reply:
I know, had the commit, was waiting the CI.